lib/block-supports/index.php

```diff
@@ -37,7 +37,7 @@ function gutenberg_apply_block_supports( $block_content, $block ) {
 
 	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
 	// If no render_callback, assume styles have been previously handled.
-	if ( ! $block_type || ! $block_type->render_callback ) {
+	if ( ! $block_type || ! $block_type->apiVersion || ! $block_type->apiVersion < 2 || ! $block_type->render_callback ) {
```

Comment:
The `apiVersion` comes from block.json and is camel case (to match js rules), how did we solve this kind of issues in previous block.json properties cc @gziolo @mcsf @jorgefilipecosta

Reply:
We map everywhere to meet PHP coding style guidelines. It should be `api_version` in the block type object instance. I'm not sure about other places though.

Reply:
So when we call `register_block_type` explicitly, we should pass `api_version` right?

Reply:
Correct, this is what `regiser_block_type_from_metadata` does for other settings:

https://github.com/WordPress/wordpress-develop/blob/3a19ff15f9013f7acf6951def3937807ea87733f/src/wp-includes/blocks.php#L197-L210

Reply:
Ok so for now, I explicitly added `api_version` to the registration call. We should do a patch for Core to add the api_version to that array.

Reply:
Right, that works, too 👍

Reply:
I agree with translating to snake case in PHP and with subsequently patching `register_block_type_from_metadata`.